models/src/controller.rs

```diff
@@ -290,3 +310,50 @@ pub fn controller_priority_class() -> PriorityClass {
         ..Default::default()
     }
 }
+
+/// Convert update time window values to cron expression
+/// For overnight schedule, we format it to `{}-23,0-{}` on hour spot.
+/// For example, start time on 18 and stop time on 5. we format it to `18-23,0-5`
+fn cron_expression_converter(update_window_start: String, update_window_stop: String) -> String {
+    let start_hour: u8 = update_window_start
+        .split(":")
+        .next()
+        .unwrap()
+        .parse()
+        .unwrap();
+    let stop_hour: u8 = update_window_stop
+        .split(":")
+        .next()
+        .unwrap()
+        .parse()
+        .unwrap();
+
+    let cron_expression_hour = if start_hour <= stop_hour {
+        format!("* * {}-{} * * * *", start_hour, stop_hour)
+    } else {
+        format!("* * {}-23,0-{} * * * *", start_hour, stop_hour)
+    };
+
+    cron_expression_hour
+}
+#[cfg(test)]
+pub(crate) mod test {
+    use super::*;
+
+    #[tokio::test]
+    async fn test_cron_expression_converter() {
+        let test_cases = vec![
+            ("0:0:0", "5:0:0", "* * 0-5 * * * *"),
+            ("21:0:0", "8:30:0", "* * 21-23,0-8 * * * *"),
+            ("15:0:0", "3:30:34", "* * 15-23,0-3 * * * *"),
+            ("0:0:0", "0:0:0", "* * 0-0 * * * *"),
```

Comment:
So, this is _"always"_? The full 24 hours?

Reply:
Just trying to wrap my head around `0-0` in the hour part 😬 

`* * 0-0 * * * *` means 24/7?

Reply:
`* * 0-0 * * * *` means only update on `0`. `"0:0:0" to "0:0:0"` is an odd edge case that users may specify, but I don't believe it's a meaningful use case. I assume if users want to run 24/7, they just use default, so if they specify to like this, controller will recognize it as only update on `0`.